tests/plans/unit_tests/test_verify_undulator_gap.py

```diff
@@ -0,0 +1,106 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+from ophyd.sim import make_fake_device
+
+from dodal.devices.DCM import DCM
+from dodal.devices.undulator import Undulator
+from dodal.plans.verify_undulator_gap import (
+    UndulatorGapAccess,
+    _get_closest_gap_to_dcm_energy,
+    _get_energy_to_distance_table,
+    verify_undulator_gap,
+)
+
+
+@pytest.fixture
+def fake_undulator() -> Undulator:
+    undulator = make_fake_device(Undulator)(name="undulator")
+    return undulator
+
+
+@pytest.fixture
+def fake_dcm() -> DCM:
+    dcm = make_fake_device(DCM)(name="dcm")
+    return dcm
+
+
+@patch("dodal.plans.verify_undulator_gap.LOGGER")
+def test_when_gap_access_is_disabled_function_returns_false_and_logger_gives_warning(
+    mock_logger, fake_dcm: DCM, fake_undulator: Undulator
+):
+    fake_undulator.gap_access.sim_put(UndulatorGapAccess.DISABLED.value)
+    assert verify_undulator_gap(fake_undulator, fake_dcm) is False
+    mock_logger.warning.assert_called_once()
+
+
+def create_energy_distance_table(path):
+    # Create lookup text file similar to the lookup table we see on I03
+    with open(path, mode="a") as file:
+        file.write(
+            "#test file\n\
+            Units    eV    mm\n\
+            5700    5.4606\n\
+            14000   5.674\n\
+            6000    5.681"
+        )
```

Comment:
Should: better to mock out the `file.open` than to create an actual file on disk